src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -45,6 +45,7 @@ internal FakeManager(Type fakeObjectType, object proxy)
                                          new CallRuleMetadata { Rule = new ObjectMemberRule(this) },
                                          new CallRuleMetadata { Rule = new AutoFakePropertyRule(this) },
                                          new CallRuleMetadata { Rule = new PropertySetterRule(this) },
+                                         new CallRuleMetadata { Rule = new CancellationRule() },

```

Comment:
What if a property getter has a cancellation token as an argument? Do we want the property-ness behaviour to be preserved, or the cancellation behaviour?
I'd originally imagined the cancellation rule being first in the list.